File: src/3340.check-balanced-string.3636/solution.java
class Solution {
    public boolean isBalanced(String num) {
        int diff = 0;
        char[] chars = num.toCharArray();
        for(int i=0;i<chars.length;i++){
            int d = chars[i]-'0';
            if(i%2 == 0){
                diff += d;
            }else{
                diff -= d;
            }
        }
        return diff == 0;
    }
}

File: src/3340.check-balanced-string.3636/solution.ts
function isBalanced (num: string): boolean {
    let diff = 0;
    for (let i = 0; i < num.length; i++) {
        const n = +num[i];
        if (i % 2 === 0) {
            diff += n;
        } else {
            diff -= n;
        }
    }
    return diff === 0;
}
